sys/gen/pmap.c

```diff
@@ -442,6 +441,41 @@ int pmap_emulate_bits(pmap_t *pmap, vaddr_t va, vm_prot_t prot) {
   return 0;
 }
 
+int pmap_fault_handler(ctx_t *ctx, vaddr_t vaddr, vm_prot_t access) {
+  thread_t *td = thread_self();
+  int error = EINVAL;
+
+  if (kern_addr_p(vaddr))
+    goto fault;
+
+  pmap_t *pmap = pmap_user();
+  assert(pmap);
+
+  if (!(error = pmap_emulate_bits(pmap, vaddr, access)))
+    return 0;
+
+  if (error == EACCES)
+    goto fault;
+
+  vm_map_t *vmap = vm_map_user();
+  assert(vmap);
```

Comment:
It's already asserted in `vm_map_user` body.

Reply:
Right. This line is obsolete. Done.